src/libponyrt/actor/actor.c

```diff
@@ -90,6 +90,22 @@ static void send_unblock(pony_actor_t* actor)
   ponyint_cycle_unblock(actor);
 }
 
+static void send_block(pony_ctx_t* ctx, pony_actor_t* actor)
+{
+  // We're blocked, send block message.
+  set_flag(actor, FLAG_BLOCKED_SENT);
+  set_flag(actor, FLAG_CD_CONTACTED);
+  pony_assert(ctx->current == actor);
```

Comment:
I moved the assertion up a few lines. i kept it because the original code path in `handle_message` had it and it's only asserted in a debug build so doesn't impact performance in release builds. The reason not use use `ctx->current` when the same value is in `actor` is because `actor` should already be available as a value in a register while loading from `ctx->current` could result in requiring a memory load. Note: the former may be incorrect but that is my limited understanding of how compilers tend to translate C code to assembly.